queries/period_slippage.sql

```diff
@@ -310,70 +310,114 @@ incoming_and_outgoing_with_buffer_trades as (
     from buffer_trades
 ),
 final_token_balance_sheet as (
-    select solver_address,
-           solver_name,
-           sum(amount) token_imbalance_wei,
-           symbol,
-           token,
-           tx_hash
-    from incoming_and_outgoing_with_buffer_trades
-    group by symbol,
-             token,
-             solver_address,
-             solver_name,
-             tx_hash
+    select
+        solver_address,
+        solver_name,
+        sum(amount) token_imbalance_wei,
+        symbol,
+        token,
+        tx_hash,
+        date_trunc('minute', block_time) as minute
+    from
+        incoming_and_outgoing_with_buffer_trades
+    group by
+        symbol,
+        token,
+        solver_address,
+        solver_name,
+        tx_hash,
+        block_time
+    having
+        sum(amount) != 0
 ),
-
-exact_prices as (
+token_times as (
     select
-        token as contract_address,
-        decimals,
-        price
-    from final_token_balance_sheet
-    left join prices.usd
-        on contract_address = token
-        and minute = '{{EndTime}}'
+        date_trunc('hour', minute) as hour,
+        minute,
+        token
+    from
+        final_token_balance_sheet
+    group by
+        hour,
+        minute,
+        token
 ),
--- The approximate prices are based on the hourly median prices over the past day.
-approximate_prices as (
+token_times_hour as (
     select
-        contract_address,
-        decimals,
-        percentile_cont(0.5) WITHIN group (order by median_price) as price
-    from prices.prices_from_dex_data p
-    where hour > '{{EndTime}}'::timestamptz - interval '1 day'
-    and date(hour) = '{{EndTime}}'::timestamptz - interval '1 day'
-      -- Getting approximate prices only for tokens we do not have an exact price for.
-    and contract_address IN (select contract_address from exact_prices where price is null)
-    group by contract_address, decimals
-    having sum(sample_size) > 0
+        hour,
+        token
+    from
+        token_times
+    group by
+        hour,
+        token
 ),
-end_prices as (
-    select * from exact_prices where price is not null
-    union
-    select * from approximate_prices
+precise_prices as (
+    select
+        pusd.*
+    from
+        prices.usd pusd
+        inner join token_times tt on pusd.minute = tt.minute
+        and contract_address = token
+),
+median_prices as (
+    select
+        musd.*
+    from
+        prices.prices_from_dex_data musd
+        inner join token_times_hour tt on musd.hour = tt.hour
```

Comment:
Usually, we have sample size >0, but this is given by default, since cowswap traded on it and price data considers cowswap trades to my knowledge. Admittedly this self reference is ugly :) 
I did not consider using the day price, as I understood that we want to have the price of auction. But not sure. Do you feel strongly to change it?

Reply:
If it's an internalized intermediate hop we won't have a DEX trade would we? You said you checked for outliers? In this case I'd be fine leaving it without a threshold, but I think it's risky and may lead to similar problems as in the 6 ETH miscomputation.

Reply:
I am testing to replace it by:
```
median_prices as (
    select
        token as contract_address,
        hour, 
        (Select percentile_cont(0.5) WITHIN group (order by median_price) as median_price
            from prices.prices_from_dex_data p
            where p.contract_address = token
            and tt.hour > p.hour - interval '12 hours'
            and tt.hour <= p.hour
            -- Getting approximate prices only for tokens we do not have an exact price for.
            and contract_address IN (select contract_address from precise_prices where price is null)
            group by contract_address, decimals
            having sum(sample_size) > 0),
        (Select decimals
            from prices.prices_from_dex_data p
            where p.contract_address = token
            limit 1)
        from token_and_time_of_trading_hourly tt
),
```
But the performance really suffers from it. The script takes then 7 minutes to execute

Reply:
If we go for this change, then we basically get the same payout output as before:
```
2022-06-15 08:50:37,530 INFO duneapi.api Fetching ETH Reimbursement & COW Rewards on Ethereum Mainnet...
2022-06-15 08:50:43,572 INFO duneapi.api got 38 records from last query
2022-06-15 08:50:43,787 INFO duneapi.api Fetching Updated Token List on Ethereum Mainnet...
2022-06-15 08:50:44,858 INFO duneapi.api got 51 records from last query
2022-06-15 08:50:44,858 INFO duneapi.api Fetching Slippage Accounting on Ethereum Mainnet...
2022-06-15 08:52:20,925 INFO duneapi.api got 18 records from last query
2022-06-15 08:52:20,930 INFO duneapi.api Fetching Solver Reward Targets on Ethereum Mainnet...
2022-06-15 08:52:27,273 INFO duneapi.api got 23 records from last query
Redirecting solver 0x080a8b1E2f3695E179453c5e617b72A381BE44B9 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Redirecting solver 0x0d2584DA2F637805071f184bCFa1268eBAe8a24A COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Redirecting solver 0x109BF9E0287Cc95cc623FBE7380dD841d4bdEb03 COW tokens to 0x84dbAE2549d67CaF00f65C355de3D6F4df59A32C
Adjusting 0x15F4c337122Ec23859EC73Bec00aB38445E45304(prod-ParaSwap) transfer by -2.13445 (slippage)
Redirecting solver 0x15F4c337122Ec23859EC73Bec00aB38445E45304 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Redirecting solver 0x2d15894FaC906386fF7f4Bd07fcEAc43fcF80C73 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Adjusting 0x340185114F9D2617DC4c16088d0375D09fee9186(prod-Naive) transfer by -0.00227 (slippage)
Redirecting solver 0x340185114F9D2617DC4c16088d0375D09fee9186 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Adjusting 0x5B0BFe439ab45A4F002C259b1654ED21c9a42D69(barn-PLM) transfer by -0.17625 (slippage)
Redirecting solver 0x5B0BFe439ab45A4F002C259b1654ED21c9a42D69 COW tokens to 0xf7f9C5ecAFd159271fe55A8FEdEDa95d7fcD0DCc
Redirecting solver 0x6fa201C3Aff9f1e4897Ed14c7326cF27548d9c35 COW tokens to 0x84dbAE2549d67CaF00f65C355de3D6F4df59A32C
Redirecting solver 0x77ec2A722c2393D3fD64617BBaF1499C713e616b COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Redirecting solver 0x833f076D182123cA8DDE2743045Ea02957bD61ef COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Redirecting solver 0x8ccc61dBA297833dbe5D95FD6360f106b9A7576e COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Redirecting solver 0xa97DEF4fBCBa3b646dd169Bc2eeE40f0f3fE7771 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Redirecting solver 0xda324c2f06d3544E7965767ce9ca536dcb67a660 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Redirecting solver 0xdAE69AffE582d36f330Ee1145995A53Fab670962 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Adjusting 0xdE1c59Bc25D806aD9DdCbe246c4B5e5505645718(prod-1inch) transfer by -5.16199 (slippage)
Redirecting solver 0xdE1c59Bc25D806aD9DdCbe246c4B5e5505645718 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Redirecting solver 0xDe786877a10DBb7EBa25a4DA65aEcf47654F08ab COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Adjusting 0xe33062a24149f7801a48B2675Ed5111d3278F0f5(barn-1inch) transfer by -0.04312 (slippage)
Redirecting solver 0xe33062a24149f7801a48B2675Ed5111d3278F0f5 COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
Adjusting 0xE58c68679e7AAb8ef83BF37E88b18eB1f6E30e22(prod-PLM) transfer by -0.38290 (slippage)
Redirecting solver 0xE58c68679e7AAb8ef83BF37E88b18eB1f6E30e22 COW tokens to 0xf7f9C5ecAFd159271fe55A8FEdEDa95d7fcD0DCc
Adjusting 0xe92F359e6F05564849AFa933CE8F62b8007A1d5D(prod-0x) transfer by -1.29031 (slippage)
Redirecting solver 0xe92F359e6F05564849AFa933CE8F62b8007A1d5D COW tokens to 0x6C642caFCbd9d8383250bb25F67aE409147f78b2
dumping 22 results to transfers-2022-06-07-to-2022-06-14.csv
Total ETH Funds needed: 87.63401129022937
Total COW Funds needed: 400800.0
For solver payouts, paste the transfer file CSV Airdrop at:
https://gnosis-safe.io/app/eth:0xA03be496e67Ec29bC62F01a428683D7F9c204930/apps?appUrl=https://cloudflare-ipfs.com/ipfs/Qme49gESuwpSvwANmEqo34yfCkzyQehooJ5yL7aHmKJnpZ/
```

Reply:
I don’t really see the point of this. Why use the median price over the previous 12 hours for one hour? Its obscure and suffers from performance.